genai_bench/auth/storage_auth_provider.py

```diff
@@ -0,0 +1,43 @@
+"""Storage authentication provider interface."""
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional
+
+
+class StorageAuthProvider(ABC):
+    """Base class for storage authentication providers."""
+
+    @abstractmethod
+    def get_client_config(self) -> Dict[str, Any]:
+        """Get storage service-specific client configuration.
+
+        Returns:
+            Dict[str, Any]: Configuration for storage client initialization
+        """
+        pass
+
+    @abstractmethod
+    def get_auth_credentials(self) -> Any:
```

Comment:
Is there a reason we use `get_auth_credentials` for `StorageAuthProvider`, but `get_credentials` for `ModelAuthProvider`?